redditdownload/imgur-downloader/imguralbum.py

```diff
@@ -0,0 +1,396 @@
+#!/usr/bin/env python3
+# encoding: utf-8
+
+
+"""
+imguralbum.py - Download a whole imgur album in one go.
+
+Provides both a class and a command line utility in a single script
+to download Imgur albums.
+
+MIT License
+Copyright Alex Gisby <alex@solution10.com>
+
+Fork Created Aug. 2016 by jtara1
+"""
+
+
+import sys
+import re
+import urllib.request, urllib.parse, urllib.error
+import os
+import math
+import time
+from collections import Counter
+
+help_message = """
+Quickly and easily download an album from Imgur.
+
+Format:
+    $ python imguralbum.py [album URL] [destination folder]
+
+Example:
+    $ python imguralbum.py http://imgur.com/a/uOOju#6 /Users/alex/images
+
+If you omit the dest folder name, the utility will create one with the same name
+as the album
+(for example for http://imgur.com/a/uOOju it'll create uOOju/ in the cwd)
+"""
+
+
+
+class ImgurException(Exception):
+    def __init__(self, msg=False):
+        self.msg = msg
+
+
+class ImgurDownloader:
```

Comment:
Ok I'll use sphinx since I think that's what I've seen in RedditImageGrab previously.